client/my-sites/checklist/wpcom-checklist/wpcom-task-list.js

```diff
@@ -2,37 +2,43 @@
 /**
  * External dependencies
  */
-import { get, memoize, omit } from 'lodash';
+import { get, memoize, omit, pick, isBoolean } from 'lodash';
 import debugModule from 'debug';
-
 import config from 'config';
 
+/**
+ * Internal dependencies
+ */
+import { getSiteTypePropertyValue } from 'lib/signup/site-type';
+import { getVerticalTaskList } from './vertical-task-list';
+
 const debug = debugModule( 'calypso:wpcom-task-list' );
 
-export default class WpcomTaskList {
-	constructor( taskStatuses, designType, isSiteUnlaunched ) {
-		this.tasks = [];
-
-		const getTask = taskId => get( taskStatuses, taskId );
-		const hasTask = taskId => getTask( taskId ) !== undefined;
-		const isCompleted = taskId => get( getTask( taskId ), 'completed', false );
-
-		const addTask = ( taskId, completedStatus = undefined ) => {
-			const task = Object.assign(
-				{},
-				{
-					id: taskId,
-					isCompleted: completedStatus !== undefined ? completedStatus : isCompleted( taskId ),
-				},
-				omit( getTask( taskId ), 'completed' )
-			);
-
-			this.tasks.push( task );
-		};
+function getTasks( { taskStatuses, designType, isSiteUnlaunched, siteSegment, siteVerticals } ) {
+	const tasks = [];
+	const segmentSlug = getSiteTypePropertyValue( 'id', siteSegment, 'slug' );
```

Comment:
I needed to add `parseInt( siteSegment )` for this to work. The ids in the [site type collection](https://github.com/Automattic/wp-calypso/blob/66f8f9a/client/lib/signup/site-type.js
) are integers.

Could we return `segment: 1` from the API perhaps?

Reply:
@ramonjd Now the API returns an integer type segment ID or null. Thanks. :)